joern-cli/frontends/swiftsrc2cpg/src/test/scala/io/joern/swiftsrc2cpg/passes/ast/ClosureWithCompilerTests.scala

```diff
@@ -0,0 +1,258 @@
+package io.joern.swiftsrc2cpg.passes.ast
+
+import io.joern.swiftsrc2cpg.testfixtures.SwiftCompilerSrc2CpgSuite
+import io.shiftleft.codepropertygraph.generated.*
+import io.shiftleft.codepropertygraph.generated.nodes.*
+import io.shiftleft.semanticcpg.language.*
+
+class ClosureWithCompilerTests extends SwiftCompilerSrc2CpgSuite {
+
+  "ClosureWithCompilerTests" should {
+
+    "create type decls and bindings correctly (local closure declaration)" in {
+      val testCode =
+        """
+          |import Foundation
+          |
+          |func main() {
+          |  // closure call is allways "function_ref": "single_apply"
+          |
+          |	 let compare = { (s1: String, s2: String) -> Bool in
+          |    return s1 > s2
+          |	 }
+          |	 let compareResult = compare("1", "2")
+          |
+          |	 var customersInLine = ["Chris", "Alex", "Ewa", "Barry", "Daniella"]
+          |	 // auto-closure example 1:
+          |	 let customerProvider = { customersInLine.remove(at: 0) }
+          |	 let customerProviderResult = customerProvider()
+          |
+          |	 // auto-closure example 2:
+          |	 let greet = {
+          |    print("Hello, World!")
+          |	 }
+          |	 greet()
+          |
+          |	 let greetUser = { (name: String) in
+          |	   print("Hey there, \(name).")
+          |	 }
+          |	 greetUser("Alex")
+          |
+          |	 let findSquare = { (num: Int) -> (Int) in
+          |    let square = num * num
+          |	   return square
+          |	 }
+          |	 let findSquareResult = findSquare(5)
+          |}""".stripMargin
```

Comment:
Please split up the different examples.

Reply:
I did not do that on purpose because every Swift compilation for such a snippet (+ its boilerplate project code) takes ~30 seconds on my Windows system. We test 5 closures here so that means 5 times 30 seconds which does not help when running the tests. On Mac and Linux it is no issue at all (~5 seconds).

Reply:
Ok, in that case keep it like this but add comments indicating why this is not split up.